src/main/java/org/terasology/launcher/config/Config.java

```diff
@@ -0,0 +1,167 @@
+        return launcherDir;
+    }
+
+    public boolean isCheckUpdatesOnLaunch() {
+        return checkUpdatesOnLaunch;
+    }
+
+    public boolean isCloseAfterGameStarts() {
+        return closeAfterGameStarts;
+    }
+
+    public boolean isCacheGamePackages() {
+        return cacheGamePackages;
+    }
+
+    public Package getSelectedPackage() {
+        return selectedPackage;
+    }
+
+    /**
+     * Provides a pre-filled {@link Builder} instance
+     * with all configurations copied from this. Use it
+     * to create a new {@link Config} instance with
+     * modified configurations.
+     *
+     * @return a pre-filled {@link Builder} instance
+     */
+    public Builder rebuilder() {
+        return new Builder(this);
+    }
+
+    /**
+     * Provides an empty {@link Builder} instance to fill
+     * configurations using builder pattern.
+     *
+     * @return an empty {@link Builder} instance
+     */
+    public static Builder builder() {
+        return new Builder();
+    }
+
+    /**
+     * Nested builder class to create {@link Config} instances
+     * using descriptive methods.
+     */
+    public static final class Builder {
+        private GameConfig gameConfig;
+        private Locale locale;
+        private transient Path launcherDir;
+        private boolean checkUpdatesOnLaunch;
+        private boolean closeAfterGameStarts;
+        private boolean cacheGamePackages;
+        private Package selectedPackage;
+
+        private Builder() { }
+
+        private Builder(final Config last) {
+            gameConfig = last.gameConfig;
+            locale = last.locale;
+            launcherDir = last.launcherDir;
+            checkUpdatesOnLaunch = last.checkUpdatesOnLaunch;
+            closeAfterGameStarts = last.closeAfterGameStarts;
+            cacheGamePackages = last.cacheGamePackages;
+            selectedPackage = last.selectedPackage;
+        }
+
+        public Builder gameConfig(final GameConfig newGameConfig) {
+            gameConfig = newGameConfig;
+            return this;
+        }
+
+        public Builder locale(final Locale newLocale) {
+            locale = newLocale;
+            return this;
+        }
+
+        public Builder launcherDir(final Path newLauncherDir) {
+            launcherDir = newLauncherDir;
+            return this;
+        }
+
+        public Builder checkUpdatesOnLaunch(final boolean value) {
+            checkUpdatesOnLaunch = value;
+            return this;
+        }
+
+        public Builder closeAfterGameStarts(final boolean value) {
+            closeAfterGameStarts = value;
+            return this;
+        }
+
+        public Builder cacheGamePackages(final boolean value) {
+            cacheGamePackages = value;
+            return this;
+        }
+
+        public Builder selectedPackage(final Package newSelectedPackage) {
+            selectedPackage = newSelectedPackage;
+            return this;
+        }
+
+        public Config build() {
```

Comment:
Since we have requirements before building the object we might want to offer a `canBuild()` method on the builder to check that `gameConfig`, `locale`, and `launcherDir` are set.